### PR TITLE
Doctor fixes

### PR DESCRIPTION
**Pull request by Shalashtein**

### CODE REVIEW

File: app/javascript/controllers/doctor_controller.js [hunk 1]
```diff
@@ -6,7 +6,26 @@ export default class extends Controller {
...
+    if (dry && dry > 0 && oil && oil > 0) {
+      document.getElementById('evaluation_form').submit()
+    } else {
+      alert('Please specify the participant\'s quota')
+    }
+  }
+
+  reject (event) {
+    let id = event.target.dataset.id
+    console.log(id)
```
Comment (wolak88) [inline app/javascript/controllers/doctor_controller.js:20]: nit: remove

File: app/views/doctor_portal/evaluations.html.haml [hunk 2]
```diff
@@ -26,32 +26,33 @@
...
-              = form.label t("views.doctor_portal.evaluation.max_amount_oil"), for: :oil, class: 'text-lg'
-              = form.number_field :oil, min: 0, max: 30
-              = form.submit t("views.doctor_portal.evaluation.submit"), class: 'hidden', id: 'approve_submit'
-          - else
-            = t("views.doctor_portal.evaluation.give_score_error") 
-      / Action Button Footer
-      .grid.grid-cols-2.w-full
-        = link_to t('views.doctor_portal.reject'), evaluate_path(id: @selected), "data-confirm" => "Are you sure you want to reject the particiapnt? This cannot be undone", method: :delete, class: 'flex flex-grow justify-center bg-red-300 rounded-xl text-gray py-4 mr-1' if @selected
-        = link_to t('views.doctor_portal.finalize'), '#', class: 'flex flex-grow justify-center bg-primary-500 rounded-xl text-gray py-4 ml-1', data: { action: 'click->doctor#evaluate' } if @selected
+          - if Appointment.find(@selected.id).state == 'done'
```
Comment (wolak88) [inline app/views/doctor_portal/evaluations.html.haml:29]: It's an antipattern to query db from views.
Please move querying into controller

**Review by wolak88 — CHANGES_REQUESTED** [review]

File: app/controllers/doctor_portal_controller.rb [hunk 3]
```diff
@@ -13,7 +13,15 @@ def appointments
...
   def evaluations
-    @selected = !params[:id].nil? ? Appointment.find(params[:id]) : @evaluations.first
+    if params[:id]
+      @selected = Appointment.find(params[:id])
+    elsif !@evaluations.empty?
+      @selected = @evaluations.first
+    end
+    if @selected
+      @done = @selected.state == 'done'
+      @risk = ['green', 'orange', 'red'].index(@selected.participant.profile.risk_calculated)
```
Comment (wolak88) [inline app/controllers/doctor_portal_controller.rb:23]: Are we sure that risk_calculated will be set when we do it for pending appointments?

File: app/controllers/doctor_portal_controller.rb [hunk 4]
```diff
@@ -67,7 +75,9 @@ def evaluate
 
   def reject
     s = Appointment.find(params[:id].to_i)
```
Comment (wolak88) [inline app/controllers/doctor_portal_controller.rb:77]: Let's not use one letter variables which doesn't tell us anything.
Also why do you have to cast parameter here?

**Review by wolak88 — CHANGES_REQUESTED** [review]

File: app/controllers/doctor_portal_controller.rb (hunk 3 above)
Comment (roxkon) [reply app/controllers/doctor_portal_controller.rb:23]: yes, i think it is calculated with algorithm inside our app for the doctor, doctor will have his own option to specify risk

**Review by wolak88 — APPROVED** [review]

**Pull request merged**